docs/getting_started.rst

```diff
@@ -383,7 +383,7 @@ Linting and auto-formatting
 
 Linting and code auto-formatting provided by Prettier and Black are run in the background automatically by default when you run the dev server.  It is a good to monitor for linting errors in the build process: while the build may complete, it will also issue warnings to the terminal.
 
-Linting and auto-formatting can also be run by the pre-commit hooks (installed earlier). If those are bypassed or not triggered, our Travis CI builds will also fail for unformatted code.
+A full set of linting and auto-formatting can also be applied by pre-commit hooks (instructions below). If those are bypassed or not triggered, our Travis CI builds will eventually fail.
```

Comment:
```suggestion
As a convenience, many developers will want to install linting and formatting plugins in their editor. Installing ESLint, Prettier, and Black, and Flake8 plugins in your editor will catch most (but not all) code-quality checks.

If desired, you can also install pre-commit hooks (instructions below) that will prevent you from locally committing any code to git which fails to pass the code quality checks. 

If you do not use any of these options, your code is likely fail our server-side checks and you will need to update the PR in order to get it merged.
```

Reply:
> Installing ESLint, Prettier, and Black, and Flake8 plugins in your editor will catch most (but not all) code-quality checks.

Maybe this can be elaborated in a separate section? I'm afraid that setting up effective linters in an editor is something that isn't very easy.

```
Setting up linting in your editor
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Some of the most important linting tools run in pre-commit are ESLint (frontend), Prettier (frontend), Black (Python), and Flake8 (Python). As an alternative to using pre-commit, you can add these linters to your editor manually.

You can install the linters in your Kolibri development environment.

The Python linters are invoked like this::

    /path/to/your/virtualenv/bin/python -m black <path-to-file>
    /path/to/your/virtualenv/bin/python -m flake8 <path-to-file>
```

Reply:
Btw how do you invoke Prettier and ESLint front an external environment? Can you share a screenshot of your editor that I can use?

Reply:
I'm hesitant to elaborate too much here because everyone has different dev environments. For example in Sublime Text one can use https://github.com/jgirardet/sublack and https://github.com/SublimeLinter/SublimeLinter-eslint

I don't quite understand your question about external environments

Reply:
What I mean is that Kolibri's dev environment (perhaps both virtualenv and nvm) has to be active when the Editor invokes some linting tool - isn't that a challenge? Or are the linters used in Editors normally completely separate from the Kolibri env?

Reply:
Maybe I'm thinking too much of an options window with a text input field where the user configures which command to be run by the editor for linting :)

Reply:
YES! These were great additions, thanks!!